lib/internal/modules/esm/loader.js

```diff
@@ -374,14 +377,18 @@ class ESMLoader {
       return module;
     };
     const ModuleJob = require('internal/modules/esm/module_job');
-    const job = new ModuleJob(
-      this, url, undefined, evalInstance, false, false);
-    this.moduleMap.set(url, undefined, job);
-    const { module } = await job.run();
+    const namespace = await onLoad.tracePromise(async () => {
+      const job = new ModuleJob(
+        this, url, undefined, evalInstance, false, false);
+      this.moduleMap.set(url, undefined, job);
+
+      const { module } = await job.run();
+      return module.getNamespace();
+    }, { parentURL: '<eval>', url });
```

Comment:
```suggestion
    }, { __proto__: null, parentURL: '<eval>', url });
```

Reply:
This will Al get rewritten a bunch whenever I get back to this, but I'll keep the proto thing in mind, thanks for the reminder. 🙂